lib/resourcemerge/appt_test.go

```diff
@@ -0,0 +1,113 @@
+package resourcemerge
+
+import (
+	"fmt"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func TestMergeDaemonSetUpdateStrategy(t *testing.T) {
+	// The merge functions set "modified" to true if pointer fields
+	// are nil. Since this test doesn't "zero" all possible pointer
+	// fields, modified will be true, so this test expects this
+	// behavior
+	daemonset_update_strategy_tests := []struct {
+		existing appsv1.DaemonSet
+		input    appsv1.DaemonSet
+
+		expectedModified bool
+		expected         appsv1.DaemonSet
+	}{
+		{
+			existing: appsv1.DaemonSet{
+				Spec: appsv1.DaemonSetSpec{
+					UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+						Type: "OldThing",
+						RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+							MaxUnavailable: &intstr.IntOrString{Type: 0, IntVal: 1, StrVal: ""},
+							MaxSurge:       &intstr.IntOrString{Type: 0, IntVal: 1, StrVal: ""},
+						},
+					},
+				},
+			},
+			input: appsv1.DaemonSet{
+				Spec: appsv1.DaemonSetSpec{
+					UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+						Type: "OldThing",
+						RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+							MaxUnavailable: &intstr.IntOrString{Type: 0, IntVal: 1, StrVal: ""},
+							MaxSurge:       &intstr.IntOrString{Type: 0, IntVal: 1, StrVal: ""},
+						},
+					},
+				},
+			},
+			//this "true" is expected true because of the nil pointers elsewhere in the DaemonSet struct
+			//it always takes new struct and sets modified to "true" in event of nil
+			expectedModified: true,
+			expected: appsv1.DaemonSet{
+				Spec: appsv1.DaemonSetSpec{
+					UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+						Type: "OldThing",
```

Comment:
Not sure if it is required but I would suggest it is better to use realistic value for Type which can be either "RollingUpdate" or "RollingUpdate". See https://github.com/kubernetes/api/blob/master/apps/v1/types.go#L500

Reply:
I picked something "synthetic" because I was just testing whether the field got updated or not -- in reality my understanding is the RollingUpdate field is only populated if the Type is "RollingUpdate". I didn't want it to be confusing having "realistic" values in a circumstance where they wouldn't make sense (e.g. having the RollingUpdate field populated when set to OnDelete, etc). 

I've added another test to the list that goes between OnDelete and Rolling update and replaced the synthetic values with realistic ones (so now I can both test that the field changes, and have it look real).  If you think it would be better from a best practice standpoint use the consts out of types.go instead of the explicit strings, I can do that too.

Reply:
sounds great, thank you for adding a test with realistic data as well. I don't see that added test though, perhaps it is still in your local repo :)

Reply:
Yes, it's in there now, sorry. I figured while I was in there I would would squash also.